mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -629,27 +643,37 @@ static void buildMatMulOpWithQuantInfo(OpBuilder &builder,
 }
 
 /// Both the tosa.avg_pool2d and unary ops use the same
-/// UnaruOpQuantizationAttr but avg_pool operator has its own builder as it
+/// UnaryOpQuantizationAttr but avg_pool operator has its own builder as it
 /// has additional parameters not part of the unary ops.
 static void
 buildAvgPool2dOpWithQuantInfo(OpBuilder &builder, OperationState &result,
                               Type outputType, Value input,
                               DenseArrayAttr kernel, DenseArrayAttr stride,
                               DenseArrayAttr pad, TypeAttr accType) {
-  result.addOperands(input);
+  const Location loc{result.location};
```

Comment:
I'm not sure `Location` is `const` correct since it is a Core IR Type: https://mlir.llvm.org/docs/Rationale/UsageOfConst/.

Reply:
Also, could this be moved closer to its first use?

Reply:
Okay, I'm not sure about the `const` suggestion now, is it an immutable type?